<?xml version="1.0"?>
<launch>
    <param name="/use_sim_time" value="true" />
    <node name="gazebo" pkg="gazebo_demos" type="bringup_gazebo.sh" args="$(find gazebo_demos)/worlds/diffcart_willowgarage.world" output="screen"/>

	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find gazebo_demos)/rviz/demo_3d.rviz"/>
	<!--
	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find gazebo_demos)/rviz/diffcart.rviz"/>
	-->
    <node pkg="tf" type="static_transform_publisher" name="laser_base_link_transform" args="0 0 0 0 0 0 base_link laser_link 100" />
    <node pkg="tf" type="static_transform_publisher" name="imu_base_link_transform" args="0 0 0 0 0 0 base_link imu_link 100" />

  	<!-- cartographer_node -->
  	<node pkg="cartographer_ros" type="cartographer_node" name="cartographer_node" 
        	args="-configuration_directory $(find gazebo_demos)/config
              	  -configuration_basename diffcart_3d.lua"
        	output="screen">
    	<remap from="imu" to="/imu_data"/>
    	<remap from="points2" to="/point_cloud_3d"/>
  	</node>

	<node name="cartographer_occupancy_grid_node" pkg="cartographer_ros"
      type="cartographer_occupancy_grid_node" args="-resolution 0.05" />

</launch>
